### ISSUE TITLE
support NixOS

### ISSUE THREAD

**Issue by Freed-Wu** [josa42/coc-lua#93]
Similar with <https://github.com/ntkme/sass-embedded-host-ruby/issues/116>. `coc-lua` will download `lua-language-server`, but any program which respects FHS need to be `patchelf`ed to work in NixOS. Why not let it use system-wide `lua-language-server`?

**Comment by Freed-Wu** [comment]
Currently, I must

```shell
ln -sf $(which lua-language-server) ~/.config/coc/extensions/coc-lua-data/lua-language-server/bin
```

To make it work.

**Comment by Sourdface** [comment]
At the very least there should be an option to use a different lua-language-server path, if not an option use the one on the system path. This is how most plugins that use language servers seem to work.

**Comment by Freed-Wu** [comment]
> At the very least there should be an option to use a different lua-language-server path, if not an option use the one on the system path. This is how most plugins that use language servers seem to work.

Yes, I notice `coc-clangd` do like you said. It use system's clangd by default. If clangd is not found in `$PATH`, popup a window to ask users if download clangd.

**Comment by gou4shi1** [comment]
+1
